src/aws.ts

```diff
@@ -504,6 +507,237 @@ export namespace $AWS {
       },
     });
   }
+
+  export const SDK: StepFunctionSDK = new Proxy<any>(
```

Comment:
What will it take to support the v3 clients? Could we use the same interface or should we make this SDKv2 to support SDKv3 in the future?

Reply:
The good thing is that the sdk v2 and v3 are both currently supported and updated by AWS. v3 is a bit tricky as every client is split into a separate package. So getting all the types would mean installing all the packages (and some way to find them as new ones are added) and then combining all types from the packages into one.

Another thing that makes this hard is the fact that the new sdk uses the format

```ts
import { S3Client } from "@aws-sdk/client-s3";
const s3 = new S3Client({});
await s3.send(new GetObjectCommand({/*input here*/}));
```

it also has a v2 compatible style


```ts
import { S3 } from "@aws-sdk/client-s3";
const s3 = new S3({});
await s3.getObject({/*input here*/});
```

so it seems that we would need to go for the v2 compatible types anyway.